mlrun/artifacts/dataset.py

```diff
@@ -289,3 +253,51 @@ def update_dataset_meta(
         iter=artifact_spec.iter,
         project=artifact_spec.project,
     )
+
+
+def upload_dataframe(df, target_path, format, src_path=None, meta_setter=None, **kw):
+    suffix = pathlib.Path(target_path).suffix
+    if not format:
+        if suffix and suffix in [".csv", ".parquet", ".pq"]:
+            format = "csv" if suffix == ".csv" else "parquet"
+        else:
+            format = "parquet"
+
+    if src_path and os.path.isfile(src_path):
+        store_manager.object(url=target_path).upload(src_path)
+        if meta_setter:
+            meta_setter(src_path)
+        return
+
+    if df is None:
+        return
+
+    if target_path.startswith("memory://"):
+        store_manager.object(target_path).put(df)
+        return
+
+    if format in ["csv", "parquet"]:
+        if not suffix:
+            target_path = target_path + "." + format
+        writer_string = "to_{}".format(format)
+        saving_func = getattr(df, writer_string, None)
+        target = target_path
+        to_upload = False
+        if "://" in target:
+            target = mktemp()
+            to_upload = True
+        else:
+            dir = os.path.dirname(target)
+            if dir:
+                os.makedirs(dir, exist_ok=True)
+
+        saving_func(target, **kw)
+        if to_upload:
+            store_manager.object(url=target_path).upload(target)
+        if meta_setter:
+            meta_setter(target)
+        if to_upload:
+            os.remove(target)
+        return
+
+    raise ValueError(f"format {format} not implemented yes")
```

Comment:
I know you only moved the code to function and didn't change it, but still 😅 
```suggestion
    raise mlrun.errors.MLRunInvalidArgumentError(f"Unknown format {format}")
```

Reply:
done